### DISCUSSION TITLE
Slider Bind Value

### DISCUSSION THREAD

**Discussion by sutaungmai** [huntabyte/shadcn-svelte#1560] [Help] [answered]
Here is my current working input tag looks like:
![code](https://github.com/user-attachments/assets/181f7b13-3fd8-4ca9-8c20-5a236f85c675)

But somehow I got this error message like this:
![image](https://github.com/user-attachments/assets/f4d6d825-d566-44df-9f5d-2dd8216ab909)

Here is its use case:
![code](https://github.com/user-attachments/assets/789b05b8-647c-46f3-ae03-ff660f7d0de0)

**Comment by huntabyte** [comment]
You can only bind to an identifier or member expression

**Reply by sutaungmai** [reply]
Can you break down a little bit. I wasn't clearly understanding.

Really appreciate for the help :)

**Reply by huntabyte** [reply]
No worries, so by trying to `bind:` `options.length` you're basically trying to tell the slider to mutate the length property of what I imagine to be an array of values.

What are you trying to accomplish with the slider exactly?

**Reply by sutaungmai** [reply]
Basically I was using the slider for increasing and decreasing password length. Here is the parent component looks like:

![code](https://github.com/user-attachments/assets/5625cc20-ae45-4b40-986b-9105339917d6)

Here is PasswordOptions component. You will see input there above the slider it works perfectly with range input:
![github](https://github.com/user-attachments/assets/14639a90-6b2c-4505-a1c2-27bd8f1c0a3d)

**Reply by huntabyte** [reply, marked as answer]
Ah, I see what you're trying to do now. Perhaps I will introduce the ability to have a single value for the slider rather than an array if it is a single-value slider.

In the meantime, until I make that update, you should be able to accomplish it like this. You will need to `bind:options` in the parent component that is providing the `options` and make the `options` prop `$bindable` in this component.

Something like this:
```svelte
<script lang="ts">
	let { options = $bindable() } = $props()
</script>

<Slider value={options.length} onValueChange={(v) => {
	const newValue = v[0]
	options.length = newValue
})} />
```

**Reply by sutaungmai** [reply]
Aha! Thanks a lot. I truly appreciate the time, effort, and care you put into assisting me. It works perfectly now. :)
